src/main/java/com/aws/greengrass/security/CryptoKeySpi.java

```diff
@@ -20,4 +21,9 @@ KeyManager[] getKeyManagers(URI privateKeyUri, URI certificateUri)
     KeyPair getKeyPair(URI privateKeyUri, URI certificateUri) throws ServiceUnavailableException, KeyLoadingException;
 
     String supportedKeyType();
+
+    default X509Certificate getCertificate(URI privateKeyUri, URI certificateUri)
```

Comment:
not clear to me that we need both inputs, since we're requesting the certificate. But let me know if you think that the private key would be required for some reason.

Reply:
Yeah, private key is needed only in PKCS provider component because that's the label used to retrieve the cert from key store.